go-controller/pkg/clustermanager/pod/allocator.go

```diff
@@ -310,3 +315,10 @@ func (a *PodAllocator) isPodReleased(nad, uid string) bool {
 func podIdAllocationName(nad, uid string) string {
 	return fmt.Sprintf("%s/%s", nad, uid)
 }
+
+func (a *PodAllocator) findIPAMClaim(network *nettypes.NetworkSelectionElement) (*ipamclaimsapi.IPAMClaim, error) {
+	if a.ipamClaimsFetcher == nil {
+		return nil, fmt.Errorf("no ipam claims fetcher")
```

Comment:
I am a bit lost here. 

If a user creates a pod with an IPAM claim reference on a network not configured to allow persistent IPs, the allocation will ignore the ipam claim reference, but the release will fail here?

Reply:
hm, no. 

This will throw an error, which will be handled by the caller (log & continue, still releasing the allocation).

Maybe it's a bit convoluted; seems it's hard to reason about. Would it be clearer if I made this check in the caller ? 

The caller does:
```golang
	hasPersistentIPs := network.IPAMClaimReference != ""
	if hasPersistentIPs {
		ipamClaim, err := a.findIPAMClaim(network)
		if err != nil {
			klog.Errorf("Failed to retrieve IPAMClaim %q but will release IPs: %v", network.IPAMClaimReference, err)
			hasPersistentIPs = false
		} else {
			hasPersistentIPs = hasPersistentIPs && ipamClaim != nil && len(ipamClaim.Status.IPs) > 0
		}
	}
```

Reply:
All good, I think a more indicative error might be better, like `not ipam claim reconciler, network does not allow persistent IPs` or something like that since `a.ipamClaimsReconciler` kinda acts as a flag of that, its like a paralell thing to `netInfo.AllowsPersistentIPs()`

Would it be useful to check for this in the `Init` method, check that if `netInfo.AllowsPersistentIPs()` is true, we actually got an IPAM reconciler just to be consistent all around?